### DISCUSSION TITLE
How to update portainer agents

### DISCUSSION THREAD

**Discussion by pippo73** [community-scripts/ProxmoxVE#246] [Q&A] [answered]
Hello 
could someone tell me how to update the portainer agent installed with the docker script?
thank you

**Comment by MickLesk** [comment, marked as answer]
portainer: 
docker stop portainer
docker rm portainer
docker pull portainer/portainer-ce:2.21.4
docker run -d -p 8000:8000 -p 9443:9443 --name=portainer --restart=always -v /var/run/docker.sock:/var/run/docker.sock -v portainer_data:/data portainer/portainer-ce:2.21.4

portainer_agent:
docker stop portainer_agent
docker rm portainer_agent
docker pull portainer/agent:2.21.4
docker run -d -p 9001:9001 --name portainer_agent --restart=always -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/docker/volumes:/var/lib/docker/volumes portainer/agent:2.21.4

**Reply by pippo73** [reply]
thank you :-)
